File: test/java/com/marsrovers/location/PositionTest.java
package com.marsrovers.location;


import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class PositionTest {

    @Test
    public void shouldMoveOnePointRight() throws Exception {

        Position position = new Position(2,2);

        Position movedPosition = position.moveX(+1);

        assertEquals(3, movedPosition.getX());

    }

    @Test
    public void shouldMoveOnePointLeft() throws Exception {

        Position position = new Position(2,2);

        Position movedPosition = position.moveX(-1);

        assertEquals(1, movedPosition.getX());

    }

    @Test
    public void shouldMoveOnePointUp() throws Exception {

        Position position = new Position(2,2);

        Position movedPosition = position.moveY(+1);

        assertEquals(3, movedPosition.getY());

    }

    @Test
    public void shouldMoveOnePointDown() throws Exception {

        Position position = new Position(2,2);

        Position movedPosition = position.moveY(-1);

        assertEquals(1, movedPosition.getY());

    }
}

File: test/java/com/marsrovers/location/PositionTest.kt
package com.marsrovers.location


import org.junit.Test

import org.junit.Assert.assertEquals

class PositionTest {

    @Test
    @Throws(Exception::class)
    fun shouldMoveOnePointRight() {

        val position = Position(2, 2)

        val movedPosition = position.moveX(+1)

        assertEquals(3, movedPosition.x.toLong())

    }

    @Test
    @Throws(Exception::class)
    fun shouldMoveOnePointLeft() {

        val position = Position(2, 2)

        val movedPosition = position.moveX(-1)

        assertEquals(1, movedPosition.x.toLong())

    }

    @Test
    @Throws(Exception::class)
    fun shouldMoveOnePointUp() {

        val position = Position(2, 2)

        val movedPosition = position.moveY(+1)

        assertEquals(3, movedPosition.y.toLong())

    }

    @Test
    @Throws(Exception::class)
    fun shouldMoveOnePointDown() {

        val position = Position(2, 2)

        val movedPosition = position.moveY(-1)

        assertEquals(1, movedPosition.y.toLong())

    }
}
